electron/libs/cli.ts

```diff
@@ -50,59 +52,86 @@ function newInstallProgressNotifier(full_name: string, notifyMainWindow: MainWin
 
 // the tea cli package is needed to open any other package in the terminal, so make sure it's installed and return the path
 async function installTeaCli() {
-  const installations = await porcelain.install("tea.xyz");
-  const teaPkg = installations.find((i) => i.pkg.project === "tea.xyz");
-  if (!teaPkg) {
-    throw new Error("could not find or install tea cli!");
+  const installations = await tea.porcelain.install("tea.xyz");
+  const install = installations.find((i) => i.pkg.project === "tea.xyz");
+  if (!install) {
+    throw new Error("installing tea/cli failed");
   }
-
-  return teaPkg.path.join("bin/tea");
+  return install;
 }
 
-export async function openPackageEntrypointInTerminal(pkg: GUIPackage) {
-  const cliBinPath = await installTeaCli();
-  // look up the entrypoint for the package again in case it changed. This makes
-  // hacking on the entrypoint more ergonomic
-  const { entrypoint } = await getPantryDetails(pkg.full_name);
-
-  let sh = "sh";
-  if (entrypoint) {
-    sh = path.join(getTeaPath(), `${pkg.full_name}/v*`, entrypoint);
-  }
-
-  const cmd = `"${cliBinPath}" --env=false +${pkg.full_name} "${sh}"`;
-  log.info(`opening package ${pkg.full_name} in terminal with command: ${cmd}`);
-
-  const scriptPath = await createCommandScriptFile(cmd);
-  try {
-    let stdout = "";
-    let stderr = "";
+const ptys: Record<string, pty.IPty> = {};
 
-    await new Promise((resolve, reject) => {
-      const child = spawn("/usr/bin/osascript", [scriptPath]);
-      child.stdout.on("data", (data) => {
-        stdout += data.toString().trim();
-      });
-      child.stderr.on("data", (data) => {
-        stderr += data.toString().trim();
-      });
+export async function openPackageEntrypointInTerminal(
+  guipkg: GUIPackage,
+  notifyMainWindow: MainWindowNotifier
+) {
+  const cli = await installTeaCli();
+  const project = guipkg.full_name; //TODO make the naming in the API consistent with libtea and tea/cli thanks
```

Comment:
Where is this not consistent?  It would cause major problems if the fully qualified project names aren't in sync.

Reply:
lib tea and tea cli both refer to property this as “project”.

The fact that neil didn't use our existing terminology or our fucking carefully curated types in this app really pisses me off.